portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -61,15 +60,26 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
 
-    List<DataComment> allComments = new ArrayList<>();
+    int numComment = 0;
+    // parse max number of comments from user input
```

Comment:
For implementation level comments, try to explain why the code's doing something if it's unusual or unintuitive rather than what it's doing. If it's unclear what the code's doing or it's long/complex, consider extracting it to a separate method:

int maxComments = parseMaxCommentsFromRequest(request);

private int parseMaxCommentsFromRequest(HSR request) {
  ...
}

that way you keep methods short and focused, and it also makes it easier to reuse logic without copying and pasting

Reply:
ok, makes sense